code/client/src/components/AddressInput.jsx

```diff
@@ -42,27 +94,48 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
   [currentWallet])
 
   useEffect(() => {
-    const existingKnownAddresses = Object.keys(knownAddresses)
-      .map((address) => ({
-        address,
-        network: knownAddresses[address]?.network
-      }))
+    const initKnownAddresses = async () => {
+      const existingKnownAddresses = Object.keys(knownAddresses)
+        .map((address) => knownAddresses[address])
 
-    const walletsNotInKnownAddresses = wallets.filter((wallet) =>
-      !existingKnownAddresses.find((knownAddress) =>
-        knownAddress.address === wallet.address && knownAddress.network === wallet.network)
-    )
+      const walletsNotInKnownAddresses = wallets.filter((wallet) =>
+        !existingKnownAddresses.find((knownAddress) =>
+          knownAddress.address === wallet.address && knownAddress.network === wallet.network)
+      )
 
-    // Init the known address entries for existing wallets.
-    walletsNotInKnownAddresses.forEach((wallet) => {
-      dispatch(walletActions.setKnownAddress({
-        label: wallet.name,
-        address: wallet.address,
-        network: wallet.network,
-        creationTime: wallet.effectiveTime,
-        numUsed: 0
+      const knownAddressesWithoutDomain = existingKnownAddresses.filter((knownAddress) =>
```

Comment:
we should probably add a time condition later to reduce queries, but it is probably fine for now

Reply:
Yeah, I agree, right now this is just bootstrap existing address that does not have a domain. We can use lookupTime or last lookupTime to limit the query.